docs/articles/documentation/extending-testcafe/custom-reporter/helper-methods.md

```diff
@@ -0,0 +1,335 @@
+---
+layout: docs
+title: Helper Methods
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/helper-methods
+---
+# Helper Methods
+
+The helper methods are used while [implementing a reporter](creating-reporter-plugin.md#implementing-reporter)
+to format a report output. TestCafe mixins these methods to the reporter.
+
+To refer to the helper methods, use `this`.
+
+This topic contains information about several groups of helper methods.
+
+* [Output Helpers](#output-helpers)
+* [Formatting Helpers](#formatting-helpers)
+* [Coloring Helper](#coloring-helper)
+* [Date-Time Helper](#datetime-helper)
+
+## Output Helpers
+
+All output helper methods are chainable: each method returns `this` so that the methods can be executed one right after the other.
+This allows you to write compact code like this:
+
+```js
+this.setIndent(4)
+    .write(testname)
+    .newline()
+    .write(result);
+```
+
+### newline
+
+Adds a new line to the report.
+
+```text
+newline () → this
+```
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    const summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.newline()
+        .write(summary)
+        .newline();
+}
+```
+
+### write
+
+Writes the specified text in the report.
+
+```text
+write (text) → this
+```
+
+Parameter | Type   | Description
+--------- | ------ | ---------------------------------
+`text`    | String | The text to output in the report.
+
+**Example**
+
+```js
+reportTaskDone (endTime, passed) {
+    ...
+    const summary = passed === this.testCount ?
+                  `${this.testCount} passed` :
+                  `${this.testCount - passed}/${this.testCount} failed`;
+
+    this.write(summary)
+        .newline();
+}
+```
+
+### useWordWrap
+
+Toggles word wrapping for subsequent output.
+
+```text
+useWordWrap (use) → this
+```
+
+Parameter | Type    | Description
+--------- | ------- | ---------------------------------------
+`use`     | Boolean | Specify *true* to enable word wrapping.
+
+**Example**
+
+The following example demonstrates how to enable word wrapping in case when the `title` string might be long.
```

Comment:
The following example demonstrates how to enable word wrapping in case the `title` string is long.